### PR TITLE
Rewrite Kafka client master and worker

### PR DESCRIPTION
**Pull request by alyiwang**
Dependency on wherehows-commons/kafka will be removed later on. Old consumerMaster and worker will also be remove after the processors have been refactored.

Tested consumer in local

### CODE REVIEW

File: wherehows-common/src/main/java/wherehows/common/Constant.java [hunk 1]
```diff
@@ -269,4 +269,10 @@
 
   // ump metrics
   public static final String UMP_METRIC_CSV_FILE_KEY = "ump.metric_csv_file";
+
+  // kafka client
+  public static final String KAFKA_JOB_TYPE = "kafka";
```
Comment (mars-lan) [inline wherehows-common/src/main/java/wherehows/common/Constant.java]: We can drop the job type now there's a separate folder for kafka jobs

File: wherehows-kafka/src/main/java/wherehows/processors/SimplePrintProcessor.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+package wherehows.processors;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import wherehows.dao.DaoFactory;
+
+
+/**
+ * Abstract class for Kafka message processor.
```
Comment (mars-lan) [inline wherehows-kafka/src/main/java/wherehows/processors/SimplePrintProcessor.java]: Update comments

File: wherehows-kafka/src/main/java/wherehows/processors/SimplePrintProcessor.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ */
+@Slf4j
+public class SimplePrintProcessor extends KafkaMessageProcessor {
```
Comment (mars-lan) [inline wherehows-kafka/src/main/java/wherehows/processors/SimplePrintProcessor.java]: Maybe call this DummyProcessor or something?

File: wherehows-kafka/src/main/java/wherehows/main/ApplicationStart.java [hunk 4]
```diff
@@ -85,7 +86,10 @@ public static void main(String[] args) {
 
     writeProcessId();
 
-    ActorSystem actorSystem = ActorSystem.create("WhereHowsKAFKAConsumerService");
+    ActorSystem actorSystem = ActorSystem.create("KAFKA");
     actorSystem.actorOf(Props.create(KafkaConsumerMaster.class), "KafkaMaster");
+
+    //String kafkaJobDir = config.getString("kafka.consumer.dir");
```
Comment (mars-lan) [inline wherehows-kafka/src/main/java/wherehows/main/ApplicationStart.java]: Remove?

File: wherehows-kafka/src/main/java/wherehows/actors/KafkaClientMaster.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    } else {
+      unhandled(message);
+    }
+  }
+
+  @Override
+  public void postStop() throws Exception {
+    log.info("Terminating KafkaClientMaster...");
+    for (ActorRef worker : _kafkaWorkers.values()) {
+      worker.tell("Shutdown", getSelf());
```
Comment (mars-lan) [inline wherehows-kafka/src/main/java/wherehows/actors/KafkaClientMaster.java]: Is this the recommended way to shutdown workers?

File: wherehows-common/src/main/java/wherehows/common/utils/JobsUtil.java [hunk 6]
```diff
@@ -73,12 +73,27 @@ private static String resolveEnviornmentalVariable(String value) {
...
     return filename.substring(0, filename.lastIndexOf('.'));
   }
 
   /**
+   * Get filename without extension
+   * @param path Path
+   * @return String
+   */
+  public static String jobNameFromPath(Path path) {
+    String filename = path.getFileName().toString();
```
Comment (mars-lan) [inline wherehows-common/src/main/java/wherehows/common/utils/JobsUtil.java]: Should be able to reuse the above function here?

File: wherehows-common/src/main/java/wherehows/common/utils/JobsUtil.java [hunk 7]
```diff
@@ -96,6 +111,23 @@ private static String jobNameFromFile(File file) {
...
+        }
+      }
+    }
+    return jobs;
+  }
+
+  /**
    * Returns a map of job name to job properties which are enabled and of certain type.
    */
   public static Map<String, Properties> getEnabledJobsByType(String dir, String type) {
```
Comment (mars-lan) [inline wherehows-common/src/main/java/wherehows/common/utils/JobsUtil.java]: Do we still need this function?

Comment (alyiwang) [reply wherehows-common/src/main/java/wherehows/common/utils/JobsUtil.java]: There is one other usage in TreeBuilderActor:    
treeJobList = JobsUtil.getEnabledJobsByType(ETL_JOBS_DIR, TREE_JOB_TYPE);

**Pull request merged**